### PR TITLE
Updated EvaluationError variants and removed .unwrap() in expressions and aggregating functions

### PR DESCRIPTION
**Pull request by ruokun-niu**
The remainings will be updated in a separate PR

### CODE REVIEW

File: core/src/evaluation/functions/aggregation/avg.rs [hunk 1]
```diff
@@ -63,7 +66,10 @@ impl AggregatingFunction for Avg {
             }
             VariableValue::Integer(n) => {
                 *count += 1;
-                *sum += n.as_i64().unwrap() as f64;
+                *sum += match n.as_i64() {
+                    Some(n) => n as f64,
```
Comment (agentofreality) [inline core/src/evaluation/functions/aggregation/avg.rs:82]: Shouldn't we be summing i64s and converting to f64 at the point of division?

File: core/src/evaluation/functions/aggregation/linear_gradient.rs [hunk 2]
```diff
@@ -205,8 +205,14 @@ impl AggregatingFunction for LinearGradient {
...
 fn extract_parameter(p: &VariableValue) -> Result<f64, EvaluationError> {
     let result = match p {
-        VariableValue::Float(n) => n.as_f64().unwrap(),
-        VariableValue::Integer(n) => n.as_i64().unwrap() as f64,
+        VariableValue::Float(n) => match n.as_f64() {
+            Some(n) => n,
+            None => return Err(EvaluationError::ConversionError),
+        }
+        VariableValue::Integer(n) => match n.as_i64() {
+            Some(n) => n as f64,
```
Comment (agentofreality) [inline core/src/evaluation/functions/aggregation/linear_gradient.rs:237]: Why is this an f64?

File: core/src/evaluation/functions/aggregation/linear_gradient.rs [hunk 3]
```diff
@@ -205,8 +205,14 @@ impl AggregatingFunction for LinearGradient {
...
+            None => return Err(EvaluationError::ConversionError),
+        },
         VariableValue::Duration(d) => d.duration().num_milliseconds() as f64,
```
Comment (agentofreality) [inline core/src/evaluation/functions/aggregation/linear_gradient.rs:243]: Not sure why these are being cast to f64s either?

File: core/src/evaluation/functions/aggregation/avg.rs (hunk 1 above)
Comment (ruokun-niu) [reply core/src/evaluation/functions/aggregation/avg.rs:82]: I think the value of `sum` is defined to be a `f64` in the accumulator. We need to convert to f64 in order to add this value to the `sum`

File: core/src/evaluation/functions/aggregation/linear_gradient.rs (hunk 2 above)
Comment (ruokun-niu) [reply core/src/evaluation/functions/aggregation/linear_gradient.rs:237]: From my understanding, the x and y values in the accumulator are `f64`.  This helper function `extract_parameter` also returns `Result<f64, EvaluationError>`

**Pull request closed without merging**